src/main/java/reposense/git/GitUtil.java

```diff
@@ -79,15 +89,51 @@ static String convertToGitFormatsArgs(List<Format> formats) {
 
     /**
      * Returns the {@code String} command to specify the globs to exclude for `git log` command.
+     * Also checks that every glob in {@code ignoreGlobList} only targets files within the given repository {@code root}
      */
-    static String convertToGitExcludeGlobArgs(List<String> ignoreGlobList) {
+    public static String convertToGitExcludeGlobArgs(File root, List<String> ignoreGlobList) {
         StringBuilder gitExcludeGlobArgsBuilder = new StringBuilder();
         final String cmdFormat = " " + addQuote(":(exclude)%s");
         ignoreGlobList.stream()
-                .filter(item -> !item.isEmpty())
+                .filter(item -> !item.isEmpty() && isValidPath(root, item))
```

Comment:
Let's add the `empty` string check inside the `isValidPath` method as well, so we don't need to have 2 different boolean value here.